lib/engine/game/g_1849/game.rb

```diff
@@ -771,12 +824,137 @@ def bank_sort(corporations)
+        end
+
+        def electric_dreams?
+          @electric_dreams ||= @optional_rules&.include?(:electric_dreams)
+        end
+
+        def bonds?
+          @bonds ||= @optional_rules&.include?(:bonds)
+        end
+
+        def reduced_4p_corps?
+          @reduced_4p_corps ||= @optional_rules&.include?(:reduced_4p_corps)
+        end
+
+        def event_buy_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:buy_tokens][1]} --"
+        end
+
+        def event_bonds!
+          @log << "-- Event: #{EVENTS_TEXT[:bonds][1]} --"
+        end
+
+        def event_e_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:e_tokens][1]} --"
+        end
+
+        # code below is for Bonds variant
+        NUM_LOANS = 6
+        MAXIMUM_LOANS = 1
+        LOAN_VALUE = 500
+        INTEREST_RATE = 50
+
+        def init_loans
+          Array.new(num_loans) { |id| Loan.new(id, loan_value) }
+        end
+
+        def num_loans
+          bonds? ? NUM_LOANS : 0
+        end
+
+        def maximum_loans(_entity)
+          MAXIMUM_LOANS
+        end
+
+        def loan_value(_entity = nil)
+          LOAN_VALUE
+        end
+
+        def interest_rate
+          INTEREST_RATE
+        end
+
+        def interest_owed_for_loans(loans)
+          interest_rate * loans
+        end
+
+        def interest_owed(entity)
+          interest_owed_for_loans(entity.loans.size)
+        end
+
+        def loans_due_interest(entity)
+          entity.loans.size
+        end
+
+        def log_interest_payment(entity, amount)
+          amount_fmt = format_currency(amount)
+          @log << "#{entity.name} pays #{amount_fmt} interest for its issued bond"
+        end
+
+        def take_loan(entity, loan)
+          raise GameError, 'Cannot issue bond' unless can_take_loan?(entity)
+
+          name = entity.name
+          @log << "#{name} issues its bond and receives #{format_currency(loan_value)}"
+          @bank.spend(loan_value, entity)
+          entity.loans << loan
+          @loans.delete(loan)
+          @round.issued_bond[entity] = true
+
+          initial_sp = entity.share_price.price
+          @stock_market.move_left(entity)
+          @log << "#{entity.name}'s share price changes from" \
+                  " #{format_currency(initial_sp)} to #{format_currency(entity.share_price.price)}"
+        end
+
+        def can_take_loan?(entity)
+          bonds? &&
+            entity.corporation? &&
+            !@round.repaid_bond[entity] &&
+            entity.loans.size < maximum_loans(entity) &&
+            !@phase.status.include?('no_bonds')
+        end
+
+        def can_pay_interest?(entity, extra_cash = 0)
+          entity.cash + extra_cash >= interest_owed(entity)
+        end
+
+        # code below is for the Electric Dreams variant
+
+        def status_array(corporation)
+          return unless corporation.e_token
+
+          ['E-Token Purchased']
```

Comment:
Maybe just `E-Token`.

Reply:
I traditionally use abilities for tracking things like this. Is the status array clearer?